staging/src/k8s.io/client-go/util/cert/pem.go

```diff
@@ -106,6 +107,64 @@ func ParsePrivateKeyPEM(keyData []byte) (interface{}, error) {
 	return nil, fmt.Errorf("data does not contain a valid RSA or ECDSA private key")
 }
 
+// ParsePublicKeysPEM is a helper function for reading an array of rsa.PublicKey or ecdsa.PublicKey from a PEM-encoded byte array.
+// Reads public keys from both public and private key files.
+func ParsePublicKeysPEM(keyData []byte) ([]interface{}, error) {
+	ok := false
+	var keys []interface{}
+	for len(keyData) > 0 {
+		var block *pem.Block
+		block, keyData = pem.Decode(keyData)
+		if block == nil {
+			break
+		}
+		// Only use PEM "PUBLIC KEY" or "PRIVATE KEY" blocks without extra headers
+		if block.Type != PublicKeyBlockType || block.Type != PrivateKeyBlockType || len(block.Headers) != 0 {
+			continue
+		}
+
+		switch block.Type {
+		case ECPrivateKeyBlockType:
+			privateKey, err := x509.ParseECPrivateKey(block.Bytes)
+			if err != nil {
+				return keys, err
+			}
+			keys = append(keys, &privateKey.PublicKey)
+		case RSAPrivateKeyBlockType:
+			privateKey, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+			if err != nil {
+				return keys, err
+			}
+			keys = append(keys, &privateKey.PublicKey)
+		case PrivateKeyBlockType:
+			privateKey, err := x509.ParsePKCS8PrivateKey(block.Bytes)
+			if err != nil {
+				return keys, err
+			}
+			if parsedRSAKey, parseSuccess := privateKey.(*rsa.PrivateKey); parseSuccess {
+				keys = append(keys, &parsedRSAKey.PublicKey)
+			} else if parsedECKey, parseSuccess := privateKey.(*ecdsa.PrivateKey); parseSuccess {
+				keys = append(keys, &parsedECKey.PublicKey)
+			} else {
+				return keys, err
+			}
```

Comment:
these parse key files from users, who have all sorts of key files, some of which even include PKIX rsa key data in a `BEGIN RSA PUBLIC KEY` block :-/

the old code tolerated that sort of insanity by just trying to decode the key types it recognizes. the stdlib tls private key parsing does something similar (https://github.com/golang/go/blob/master/src/crypto/tls/tls.go#L234 and https://github.com/golang/go/blob/master/src/crypto/tls/tls.go#L280-L297)